diff --git a/dist/core/config/vite-load.js b/dist/core/config/vite-load.js
index bccfd1b0db22c19242824c8320553bd72a2b91c6..d4d8738b4693c91e8019d245dacaea74520028ed 100644
--- a/dist/core/config/vite-load.js
+++ b/dist/core/config/vite-load.js
@@ -25,8 +25,7 @@ async function loadConfigWithVite({
     const plugins = loadFallbackPlugin({ fs, root: pathToFileURL(root) });
     server = await createMinimalViteDevServer(plugins);
     if (isRunnableDevEnvironment(server.environments[ASTRO_VITE_ENVIRONMENT_NAMES.ssr])) {
-      const environment = server.environments[ASTRO_VITE_ENVIRONMENT_NAMES.ssr];
-      const mod = await environment.runner.import(configPath);
+      const mod = await server.ssrLoadModule(configPath);
       return mod.default ?? {};
     } else {
       return {};
